Source/VMware.vSphereDSC/Classes/BaseDSC.ps1

```diff
@@ -157,6 +157,15 @@ class BaseDSC {
     <#
     .DESCRIPTION
 
+    Checks if the specified VIObject is of the specified type.
+    #>
+    [bool] IsVIObjectOfTheCorrectType($viObject, $typeAsString) {
+        return ($viObject.GetType().FullName -Match $typeAsString -or ($viObject.GetType().GetInterfaces().FullName -Match $typeAsString).Length -gt 0)
```

Comment:
Why -Match. It would be more correct to use string equals. Since this function is generic it is good to go through base types as well. Now it works with type name and interfaces but it won't work in the case of

class A
class B : A

a = new B()
IsVIObjectOfTheCorrectType "A" will return false

Reply:
Extended the method to cover the base classes of the type as well.